ricecooker/classes/files.py

```diff
@@ -189,13 +189,15 @@ def __init__(self, preset=None, language=None, default_ext=None, source_url=None
         self.source_url = source_url
 
     def set_language(self, language):
-        self.language = language
-        if isinstance(self.language, str):
-            self.language = languages.getlang(language)
-            if not self.language:
-                raise TypeError("Language {} is not found".format(language))
-        if isinstance(self.language, languages.Language):
-            self.language = self.language.code
+        """ Set self.language to internal lang. repr. code from str or Language object. """
+        if isinstance(language, str):
+            language_obj = languages.getlang(language)
```

Comment:
Alternatively, you could set this to be `language` instead of `language_obj` then set self.language under the `if isinstance(language, languages.Language)` block (not important at all, but could make code more concise)